hw/top/defs.bzl

```diff
@@ -0,0 +1,57 @@
+# Copyright lowRISC contributors (OpenTitan project).
+# Licensed under the Apache License, Version 2.0, see LICENSE for details.
+# SPDX-License-Identifier: Apache-2.0
+load("//rules/opentitan:hw.bzl", "opentitan_top")
+load("//hw/top_earlgrey/data/autogen:defs.bzl", "EARLGREY")
+load("//hw/top_darjeeling/data/autogen:defs.bzl", "DARJEELING")
+
+ALL_TOPS = [
+    EARLGREY,
+    DARJEELING,
+]
+
+ALL_TOP_NAMES = [
+    top.name
+    for top in ALL_TOPS
+]
+
+def _all_ip_names():
+    names = {}
+    for top in ALL_TOPS:
+        for ip in top.ips:
+            names[ip.name] = {}
+    return names.keys()
```

Comment:
Does this need to be sorted? Will it affect configuration hashes anywhere?

Reply:
Sorting is not necessary for this PR but I guess it doesn't hurt to do it anyway if this ever gets used in a way where the order is important.